### PR TITLE
[Android] Implemented Material3 support for ProgressBar

### LINKED ISSUE #33925: Implement Material3 support for Progressbar
When the UseMaterial3 MSBuild property is enabled, the ProgressBar control should conform to the Material 3 visual design system. However, the current implementation does not fully align with Material 3 behavior.
The ProgressBar should apply Material 3–specific styling, including correct progress colors, updated visual appearance, and proper state behaviors. It should also rely on the appropriate Material 3 theme resources to ensure visual consistency across the application.

### PR DESCRIPTION
**Pull request by SyedAbdulAzeemSF4852**
### Description of Change

- This pull request updates the handling of `ProgressBar` controls on Android to support Material3 styling when the feature is enabled. It introduces a new handler, updates the registration logic, and extends color update support for Material3 progress bars.

**Android ProgressBar Material3 Support:**

* Added a new internal handler `ProgressBarHandler2` that uses `LinearProgressIndicator` from Material3 for progress bars when Material3 is enabled. (`src/Core/src/Handlers/ProgressBar/ProgressBarHandler2.Android.cs`)
* Updated `AddControlsHandlers` in `AppHostBuilderExtensions.cs` to register `ProgressBarHandler2` instead of the default handler when Material3 is enabled, preserving the original handler otherwise.

**ProgressBar Color Customization:**

* Enhanced `ProgressBarExtensions.UpdateProgressColor` to delegate color updates to a new method for Material3 progress bars (`UpdateLinearProgressIndicatorColor`) or fall back to the native color update for standard progress bars.
* Implemented `UpdateLinearProgressIndicatorColor` for `LinearProgressIndicator`, supporting theme default fallback and custom color arrays.
* Added necessary Material3 imports to `ProgressBarExtensions.cs`.

Material Design Spec - [ProgressBar](https://m3.material.io/components/progress-indicators/overview)

### Issues Fixed
Fixes #33925

### Output
| Material2 | Material3 |
|----------|----------|
|  |  |

### CODE REVIEW

**Comment by rmarinho** [comment]
## 🤖 AI Summary


<details>
<summary>📊 <strong>Expand Full Review</strong></summary>

---

<details>
<summary><strong>🔍 Pre-Flight — Context & Validation</strong></summary>

---

<details>
<summary>📝 <strong>Review Session</strong> — <strong>Fixed formatting</strong> · <a href="https://github.com/dotnet/maui/commit/43e59e1a2a5d0690339492a3e75738be816b7b89"><code>43e59e1</code></a></summary>

---

**Issue:** #33925 - Implement Material3 support for Progressbar
**PR:** #33926 - [Android] Implemented Material3 support for ProgressBar
**Author:** SyedAbdulAzeemSF4852 (community contributor, partner/syncfusion)
**Platforms Affected:** Android only
**Labels:** platform/android, community ✨, area-controls-progressbar, partner/syncfusion, material3
**Files Changed:** 3 files (2 implementation, 0 test files)

### Issue Summary

When the `UseMaterial3` MSBuild property is enabled, the `ProgressBar` control should conform to Material 3 visual design. The current implementation doesn't use Material3-specific styling (correct colors, visual appearance, theme resources via `LinearProgressIndicator`).

### Files Changed

**Fix Files:**
- `src/Controls/src/Core/Hosting/AppHostBuilderExtensions.cs` (+12, -1) — registers `ProgressBarHandler2` when Material3 is enabled on Android, falls back to `ProgressBarHandler` otherwise
- `src/Core/src/Handlers/ProgressBar/ProgressBarHandler2.Android.cs` (+16, new) — new internal handler using `LinearProgressIndicator` from Material3
- `src/Core/src/Platform/Android/ProgressBarExtensions.cs` (+34, -0) — updated `UpdateProgressColor` to delegate to `UpdateLinearProgressIndicatorColor` for Material3 progress bars

**Test Files:** None — no tests were added for this change.

### PR Discussion Summary

**Automated review comments (copilot-pull-request-reviewer):**
| File:Line | Reviewer Says | Status |
|-----------|---------------|--------|
| `ProgressBarExtensions.cs:17` | Missing space in `if(` → should be `if (` | ⚠️ UNRESOLVED (outdated thread) |
| `ProgressBarHandler2.Android.cs:16` | Tests in `ProgressBarHandlerTests.Android.cs` cast platform view to `Android.Widget.ProgressBar`, but Material3 handler returns `LinearProgressIndicator` — tests will fail with Material3 enabled | ⚠️ UNRESOLVED |

**No other human reviewer comments.**

### Edge Cases / Concerns

- [ ] Tests: No tests added; existing `ProgressBarHandlerTests.Android.cs` uses `AProgressBar GetNativeProgressBar()` which assumes `Android.Widget.ProgressBar` — fails with Material3 handler
- [ ] The `ProgressBarHandler2` is marked `internal` with a TODO to make it public in .NET 11
- [ ] Style: minor `if(` vs `if (` formatting issue (unresolved automated review thread)
- [ ] `SetIndicatorColor([])` passing empty array to reset to theme default is undocumented behavior

### Fix Candidates

| # | Source | Approach | Test Result | Files Changed | Notes |
|---|--------|----------|-------------|---------------|-------|
| PR | PR #33926 | New `ProgressBarHandler2` using `LinearProgressIndicator`, registered via `#if ANDROID` + `RuntimeFeature.IsMaterial3Enabled` check in `AppHostBuilderExtensions`, with updated `ProgressBarExtensions.UpdateProgressColor` dispatching on type | ⏳ PENDING (Gate) | 3 files (+62/-1) | Original PR |

</details>

</details>

---

<details>
<summary><strong>🚦 Gate — Test Verification</strong></summary>

---

<details>
<summary>📝 <strong>Review Session</strong> — <strong>Fixed formatting</strong> · <a href="https://github.com/dotnet/maui/commit/43e59e1a2a5d0690339492a3e75738be816b7b89"><code>43e59e1</code></a></summary>

---

**Result:** ❌ FAILED — No Tests Exist
**Platform:** android (selected — issue is Android-only)
**Mode:** Full Verification (attempted)

### Gate Status

**Tests for issue #33925 do NOT exist.**

- No files matching `*33925*` found in:
  - `src/Controls/tests/TestCases.HostApp/Issues/`
  - `src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/`
- No device tests in `src/Core/tests/DeviceTests/Handlers/ProgressBar/` cover the Material3 path
- The existing `ProgressBarHandlerTests.Android.cs` is NOT a test for this issue (no Material3 code path coverage)

### Automated Reviewer Note

A Copilot automated reviewer flagged that:
> "The existing `ProgressBarHandlerTests.Android.cs` tests expect the platform view to be of type `Android.Widget.ProgressBar`. However, when Material3 is enabled, `ProgressBarHandler2` will return a `LinearProgressIndicator` instead. Tests will fail or won't adequately test the Material3 code path."

### Recommendation

Tests are missing for this PR. The PR author should add tests before this can be merged. The `write-tests-agent` can help create appropriate tests.

</details>

</details>

---

<details>
<summary><strong>🔧 Fix — Analysis & Comparison</strong></summary>

---

<details>
<summary>📝 <strong>Review Session</strong> — <strong>Fixed formatting</strong> · <a href="https://github.com/dotnet/maui/commit/43e59e1a2a5d0690339492a3e75738be816b7b89"><code>43e59e1</code></a></summary>

---

**Skipped** — Gate failed (no tests exist). Try-fix requires Gate to pass first.

### Fix Candidates

| # | Source | Approach | Test Result | Files Changed | Notes |
|---|--------|----------|-------------|---------------|-------|
| PR | PR #33926 | New `ProgressBarHandler2` using `LinearProgressIndicator`, registered conditionally via `RuntimeFeature.IsMaterial3Enabled` | ❌ NOT TESTED (Gate failed) | 3 files (+62/-1) | Original PR |

**Exhausted:** N/A (Gate did not pass)
**Selected Fix:** N/A

</details>

</details>

---

<details>
<summary><strong>📋 Report — Final Recommendation</strong></summary>

---

<details>
<summary>📝 <strong>Review Session</strong> — <strong>Fixed formatting</strong> · <a href="https://github.com/dotnet/maui/commit/43e59e1a2a5d0690339492a3e75738be816b7b89"><code>43e59e1</code></a></summary>

---

## ⚠️ Final Recommendation: REQUEST CHANGES

### Summary

PR #33926 implements Material3 support for the Android `ProgressBar` by introducing a new `ProgressBarHandler2` that uses `LinearProgressIndicator`. The implementation follows established patterns correctly, but **no tests were added** for the new Material3 code path. Gate failed because no tests exist for issue #33925.

---

### Root Cause (Issue #33925)

When `UseMaterial3` is enabled, the `ProgressBar` handler was still creating an `Android.Widget.ProgressBar` instead of `Google.Android.Material.ProgressIndicator.LinearProgressIndicator`. This resulted in incorrect Material3 visual appearance (colors, shape, states) since the legacy widget doesn't use Material3 theme resources.

---

### Gate Result

**❌ GATE FAILED — No Tests Exist**

- No UI test files (`IssueXXXXX.cs`) added to `TestCases.HostApp` or `TestCases.Shared.Tests`
- No device tests added to `Core.DeviceTests/Handlers/ProgressBar/`
- Existing `ProgressBarHandlerTests.Android.cs` does NOT cover the Material3 code path

---

### Fix Quality Assessment

**Approach (PR's fix):**
- New internal `ProgressBarHandler2` extending `ProgressBarHandler`, overriding `CreatePlatformView()` to return `LinearProgressIndicator(MauiMaterialContextThemeWrapper.Create(Context))`
- Registered via `AppHostBuilderExtensions.AddControlsHandlers()` under `#if ANDROID` + `RuntimeFeature.IsMaterial3Enabled` runtime check
- `ProgressBarExtensions.UpdateProgressColor` dispatches via `is LinearProgressIndicator` type check, calling `UpdateLinearProgressIndicatorColor` or falling back to the original path

**Pattern consistency:** ✅ Follows the established `RuntimeFeature.IsMaterial3Enabled` + `MauiMaterialContextThemeWrapper.Create()` pattern (matches CheckBox implementation)

---

### Code Review Findings

#### 🔴 Critical Issues

**1. No Tests Added**
- No tests for Material3 ProgressBar behavior (correct `LinearProgressIndicator` creation, color updates, reset to theme default)
- The automated reviewer correctly flagged that existing `ProgressBarHandlerTests.Android.cs` uses `AProgressBar GetNativeProgressBar()` which casts to `Android.Widget.ProgressBar` — this will fail or silently skip when `ProgressBarHandler2` is active
- Use `write-tests-agent` to add appropriate device tests

#### 🟡 Suggestions

**2. `SetIndicatorColor([])` behavior relies on undocumented API behavior**
- Comment says "passing empty array resets to `colorPrimary` when length == 0" — this is implementation-specific Material3 behavior that may change. Consider using a null/default check against Material3 API documentation rather than relying on this side-effect.

**3. Title should use present imperative tense**
- Current: `[Android] Implemented Material3 support for ProgressBar`
- Suggested: `[Android] ProgressBar: Add Material3 support using LinearProgressIndicator`

**4. PR Description missing NOTE block**
- The required NOTE block for artifact testing is absent

#### ✅ Looks Good

- Handler inheritance from `ProgressBarHandler` is correct — inherits `MapProgress` and `MapProgressColor` mapper correctly
- `MauiMaterialContextThemeWrapper.Create(Context)` correctly applies Material3 theming
- `internal` visibility with TODO for .NET 11 is appropriate for incremental Material3 rollout
- `#if ANDROID` guard in `AppHostBuilderExtensions.cs` is correct
- Runtime check using `RuntimeFeature.IsMaterial3Enabled` is the established pattern
- Color null check correctly resets to theme default

---

### Title & Description Review

**Current Title:** `[Android] Implemented Material3 support for ProgressBar`
**Suggested Title:** `[Android] ProgressBar: Add Material3 support using LinearProgressIndicator`

**Description Assessment:** Good structure with "Description of Change" and "Issues Fixed" sections, includes before/after screenshots. Missing the required NOTE block.

**Required addition to top of description:**
```
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!
```

---

### Required Changes Before Merge

1. **Add tests** (critical) — Add device tests or UI tests for:
   - `ProgressBarHandler2` creates `LinearProgressIndicator` when Material3 is enabled
   - Color update correctly calls `SetIndicatorColor`
   - Null color resets to theme default
2. **Add NOTE block** to PR description
3. Optional: Update title to use present imperative tense

</details>

</details>

---

</details>



<details>
<summary>📋 <strong>Expand PR Finalization Review</strong></summary>

---

<details>

<summary><b>Title: ✅ Good</b></summary>

<br>

**Current:** `[Android] Implemented Material3 support for ProgressBar`
</details>

<details>

<summary><b>Description: ✅ Good</b></summary>

<br>

Description needs updates. See details below.
### ✨ Suggested PR Description


> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

### Description of Change

- This pull request updates the handling of `ProgressBar` controls on Android to support Material3 styling when the feature is enabled. It introduces a new handler, updates the registration logic, and extends color update support for Material3 progress bars.

**Android ProgressBar Material3 Support:**

* Added a new internal handler `ProgressBarHandler2` that uses `LinearProgressIndicator` from Material3 for progress bars when Material3 is enabled. (`src/Core/src/Handlers/ProgressBar/ProgressBarHandler2.Android.cs`)
  * `ProgressBarHandler2` is currently `internal` and is planned to be made public in .NET 11 (`// TODO: Material3 - make it public in .net 11`).
* Updated `AddControlsHandlers` in `AppHostBuilderExtensions.cs` to register `ProgressBarHandler2` instead of the default handler when Material3 is enabled on Android, preserving the original `ProgressBarHandler` otherwise.

**ProgressBar Color Customization:**

* Enhanced `ProgressBarExtensions.UpdateProgressColor` to delegate color updates to a new method for Material3 progress bars (`UpdateLinearProgressIndicatorColor`) or fall back to the native color update for standard progress bars.
* Implemented `UpdateLinearProgressIndicatorColor` for `LinearProgressIndicator`, supporting theme default fallback (via `SetIndicatorColor([])`) and custom color arrays.
* Added necessary Material3 imports to `ProgressBarExtensions.cs`.

Material Design Spec - [ProgressBar](https://m3.material.io/components/progress-indicators/overview)

### Issues Fixed
Fixes #33925

### Output
| Material2 | Material3 |
|----------|----------|
|  |  |
</details>

<details>

<summary><b>Code Review: ✅ Passed</b></summary>

<br>

# Code Review Findings — PR #33926

## 🟡 Suggestions

---

### 1. No Tests for the Material3 Code Path

**File:** `src/Core/src/Handlers/ProgressBar/ProgressBarHandler2.Android.cs` (line 16)

**Problem:** The existing `ProgressBarHandlerTests.Android.cs` does not cover the Material3 code path. There are no tests that verify:
- `ProgressBarHandler2` creates a `LinearProgressIndicator`
- Color updates via `UpdateLinearProgressIndicatorColor` work correctly (both null/default and custom colors)
- Progress value is correctly applied when using `LinearProgressIndicator`

Note: `LinearProgressIndicator` *does* extend `Android.Widget.ProgressBar` (via `BaseProgressIndicator`), so existing test helpers that cast to `AProgressBar` will not throw. However, they won't exercise the Material3 color path.

**Recommendation:** Add device tests that enable Material3, instantiate `ProgressBar`, and verify the platform view is a `LinearProgressIndicator` and color updates behave correctly.

This issue was flagged in an open (unresolved) review thread by `copilot-pull-request-reviewer`.

---

### 2. Missing Newline at End of File

**File:** `src/Core/src/Handlers/ProgressBar/ProgressBarHandler2.Android.cs`

**Problem:** The diff shows `\ No newline at end of file` — the file is missing the standard trailing newline.

**Recommendation:** Add a newline after the closing `}` on the last line of the file.

---

### 3. Open Review Thread Not Explicitly Resolved

**Thread:** Code style — missing space in `if(` vs `if (`

**Status:** The thread is marked as outdated (code was changed), and the current diff shows `if (platformProgressBar is LinearProgressIndicator materialProgressBar)` with a proper space. However, the thread is not explicitly marked as resolved on GitHub.

**Recommendation:** Author should resolve the thread to confirm the fix was applied intentionally.

---

## ✅ Looks Good

- **Handler pattern:** `ProgressBarHandler2` correctly follows the established Material3 handler pattern in the codebase — subclasses the standard handler, overrides `CreatePlatformView()`, and uses `MauiMaterialContextThemeWrapper.Create(Context)` for proper Material3 theming.
- **Registration logic:** The conditional handler registration in `AppHostBuilderExtensions.cs` is correctly scoped inside `#if ANDROID`, ensuring no impact on other platforms.
- **Color null-handling:** `UpdateLinearProgressIndicatorColor` correctly handles `null` color by calling `SetIndicatorColor([])` (empty array), which tells Material3 to resolve to the theme's `colorPrimary`. This is the correct pattern per the Material3 API.
- **No API surface changes:** `ProgressBarHandler2` is `internal`, so there are no public API changes requiring `PublicAPI.Unshipped.txt` updates.
- **Non-breaking:** On non-Android platforms and when Material3 is disabled, the original `ProgressBarHandler` is used, preserving existing behavior.
- **Covariant return type:** `ProgressBarHandler2.CreatePlatformView()` returns `LinearProgressIndicator`, which is a subtype of `Android.Widget.ProgressBar` (via `BaseProgressIndicator<T> : ProgressBar`), so the override is type-safe.
</details>

---

</details>

File: src/Core/src/Platform/Android/ProgressBarExtensions.cs [hunk 1]
```diff
@@ -14,6 +15,18 @@ public static void UpdateProgress(this AProgressBar platformProgressBar, IProgre
 		}
 
 		public static void UpdateProgressColor(this AProgressBar platformProgressBar, IProgress progress)
+		{
+			if(platformProgressBar is LinearProgressIndicator materialProgressBar)
```
[the remark this answers is not in the document]
Comment (SyedAbdulAzeemSF4852) [reply src/Core/src/Platform/Android/ProgressBarExtensions.cs]: Fixed the formatting as suggested

**Comment by SyedAbdulAzeemSF4852** [comment]
Added UI tests and addressed all valid concerns.

File: src/Core/src/Handlers/ProgressBar/ProgressBarHandler2.Android.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+{
+	protected override LinearProgressIndicator CreatePlatformView()
+	{
+		return new LinearProgressIndicator(MauiMaterialContextThemeWrapper.Create(Context))
+		{
+			Indeterminate = false,
+			Max = ProgressBarExtensions.Maximum
+		};
+	}
+}
```
[the remark this answers is not in the document]
Comment (SyedAbdulAzeemSF4852) [reply src/Core/src/Handlers/ProgressBar/ProgressBarHandler2.Android.cs]: Invalid concern.

**Review by jfversluis — COMMENTED** [review]
## Multi-Model Code Review (Sonnet 4.5 / GPT 5.1)

---

### 🔴 Type Incompatibility — LinearProgressIndicator vs Android.Widget.ProgressBar
**Severity: Critical** | Found by: GPT 5.1

`ProgressBarHandler` is `ViewHandler<IProgress, AndroidProgressBar>` where `AndroidProgressBar` = `Android.Widget.ProgressBar`. `ProgressBarHandler2` subclasses it and overrides `CreatePlatformView()` to return `LinearProgressIndicator`.

**Problem:** `LinearProgressIndicator` derives from `BaseProgressIndicator` → `android.view.View`. It does **NOT** derive from `Android.Widget.ProgressBar`. This means:

1. The override return type is not covariant with the base
2. `ViewHandler<IProgress, AndroidProgressBar>` expects an `Android.Widget.ProgressBar` as `PlatformView`
3. The `is LinearProgressIndicator` check in `UpdateProgressColor` extension is unreachable since the platform view type is `AndroidProgressBar`

**Fix:** Either:
- Change `ProgressBarHandler2` to be a standalone `ViewHandler<IProgress, LinearProgressIndicator>` (not subclassing `ProgressBarHandler`), OR
- Use a shared base type (`Android.Views.View`) and update all extension methods accordingly

---

### 🟠 Missing Track/Background Color Support
**Severity: High** | Found by: Sonnet 4.5

`UpdateLinearProgressIndicatorColor` only calls `SetIndicatorColor()`, but does NOT call `SetTrackColor()`. The test `Material3ProgressBar_SetProgressColorAndBackgroundColor_VerifyVisualState` sets both colors, but BackgroundColor changes will be applied to the container view rather than the progress track.

**Fix:** Add `SetTrackColor()` call to handle the `BackgroundColor` property correctly for `LinearProgressIndicator`.

---

### 🟡 Indeterminate Mode Hardcoded to False
**Severity: Medium** | Found by: Sonnet 4.5

`CreatePlatformView()` sets `Indeterminate = false` with no mechanism to change it at runtime. While MAUI's `IProgress` doesn't expose indeterminate mode, this should be documented as an intentional limitation.

**Comment by SyedAbdulAzeemSF4852** [comment]
> ## Multi-Model Code Review (Sonnet 4.5 / GPT 5.1)
> ### 🔴 Type Incompatibility — LinearProgressIndicator vs Android.Widget.ProgressBar
> **Severity: Critical** | Found by: GPT 5.1
> 
> `ProgressBarHandler` is `ViewHandler<IProgress, AndroidProgressBar>` where `AndroidProgressBar` = `Android.Widget.ProgressBar`. `ProgressBarHandler2` subclasses it and overrides `CreatePlatformView()` to return `LinearProgressIndicator`.
> 
> **Problem:** `LinearProgressIndicator` derives from `BaseProgressIndicator` → `android.view.View`. It does **NOT** derive from `Android.Widget.ProgressBar`. This means:
> 
> 1. The override return type is not covariant with the base
> 2. `ViewHandler<IProgress, AndroidProgressBar>` expects an `Android.Widget.ProgressBar` as `PlatformView`
> 3. The `is LinearProgressIndicator` check in `UpdateProgressColor` extension is unreachable since the platform view type is `AndroidProgressBar`
> 
> **Fix:** Either:
> 
> * Change `ProgressBarHandler2` to be a standalone `ViewHandler<IProgress, LinearProgressIndicator>` (not subclassing `ProgressBarHandler`), OR
> * Use a shared base type (`Android.Views.View`) and update all extension methods accordingly
> 
> ### 🟠 Missing Track/Background Color Support
> **Severity: High** | Found by: Sonnet 4.5
> 
> `UpdateLinearProgressIndicatorColor` only calls `SetIndicatorColor()`, but does NOT call `SetTrackColor()`. The test `Material3ProgressBar_SetProgressColorAndBackgroundColor_VerifyVisualState` sets both colors, but BackgroundColor changes will be applied to the container view rather than the progress track.
> 
> **Fix:** Add `SetTrackColor()` call to handle the `BackgroundColor` property correctly for `LinearProgressIndicator`.
> 
> ### 🟡 Indeterminate Mode Hardcoded to False
> **Severity: Medium** | Found by: Sonnet 4.5
> 
> `CreatePlatformView()` sets `Indeterminate = false` with no mechanism to change it at runtime. While MAUI's `IProgress` doesn't expose indeterminate mode, this should be documented as an intentional limitation.


@jfversluis , 
**Concern 1 — Type Incompatibility**: This concern is invalid — [LinearProgressIndicator](https://github.com/material-components/material-components-android/blob/534a3f8b66075c4589325228614bc8150d3e0894/lib/java/com/google/android/material/progressindicator/LinearProgressIndicator.java#L61) extends [BaseProgressIndicator](https://github.com/material-components/material-components-android/blob/534a3f8b66075c4589325228614bc8150d3e0894/lib/java/com/google/android/material/progressindicator/BaseProgressIndicator.java#L74-L75) which itself extends android.widget.ProgressBar, making it a proper subtype of Android.Widget.ProgressBar, so the covariant override, the is LinearProgressIndicator runtime check, and the subclassing approach in this PR are all valid.

**Concern 2 — Missing Track/Background Color Support:** This concern is invalid — UpdateLinearProgressIndicatorColor is only responsible for ProgressColor (from IProgress), while BackgroundColor is a separate IView property handled generically by ViewHandler.ViewMapper, identical to how Material2 behaves, and since IProgress has no TrackColor property, adding SetTrackColor() would require a new API — making this a future enhancement request, not a bug in this PR.

**Concern 3 — Indeterminate Mode:** This concern is invalid — Indeterminate = false is set in both the existing [ProgressBarHandler](https://github.com/dotnet/maui/blob/b3da1785d62ddefde397d5385419afb742ae1bcf/src/Core/src/Handlers/ProgressBar/ProgressBarHandler.Android.cs#L8-L14) (Material2) and the new ProgressBarHandler2 (Material3), making this identical pre-existing behavior, not a new limitation introduced by this PR.

**Review by jfversluis — DISMISSED** [review]

**Comment by jfversluis** [comment]
Something went wrong here somehow. I will fix it :)
